app/api/v2/handlers/payload_api.py

```diff
@@ -22,22 +21,124 @@ def __init__(self, services):
+        PayloadSchema(),
+        description="The created payload in a list in PayloadSchema format (with name changed in case of a duplicate).")
+    async def post_payloads(self, request: web.Request):
+        # As aiohttp_apispec.form_schema already calls request.multipart(),
+        # accessing the file using the prefilled request["form"] dictionary.
+        field_field: web.FileField = request["form"]["payload"]
+
+        file_name, file_path = await self.__generate_file_name_and_path(field_field)
+
+        # The field_field.file is of type IOBase: It uses blocking methods.
+        # Putting blocking code into a dedicated method and thread...
+        loop: asyncio.AbstractEventLoop = asyncio.get_event_loop()
+        await loop.run_in_executor(None, self.__save_file, file_path, field_field.file)
+
+        body: dict[list[str]] = {"payloads": [file_name]}
+        return web.json_response(body)
+
+    @aiohttp_apispec.docs(
+        tags=['payloads'],
+        summary='Delete a payload',
+        description='Deletes a given payload.',
+        responses = {
+            204: {"description": "Payload has been properly deleted."},
+            404: {"description": "Payload not found."},
+        })
+    @aiohttp_apispec.match_info_schema(PayloadDeleteRequestSchema)
+    async def delete_payloads(self, request: web.Request):
+        file_name: str = request.match_info.get("name")
+        file_path: str = os.path.join('data/payloads/', file_name)
+
+        response: web.HTTPException = None
+        try:
+            os.remove(file_path)
+            response = web.HTTPNoContent()
+        except FileNotFoundError:
+            response = web.HTTPNotFound()
+        return response
+
+    @classmethod
+    async def __generate_file_name_and_path(cls, field_field: web.FileField) -> [str, str]:
+        """
+        Finds whether an uploaded file already exists in the payload directory.
+        In the case, generates a new file name with an incremental suffix to avoid overriding the existing one.
+        Otherwise, the original file name is used.
+
+        :param field_field: The upload payload object.
+        :return: A tuple containing the generated file name and path for future storage.
+        """
+        file_name_candidate: str = field_field.filename
+        file_path: str = os.path.join('data/payloads/', file_name_candidate)
+        suffix: int = 1
+
+        # Generating a file suffix in the case it already exists.
+        while os.path.exists(file_path):
+            file_name_candidate = f"{pathlib.Path(field_field.filename).stem}_" \
+                                  f"{suffix}{pathlib.Path(field_field.filename).suffix}"
+            file_path = os.path.join('data/payloads/', file_name_candidate)
+            suffix += 1
+        file_name: str = file_name_candidate
+        return file_name, file_path
+
+    @staticmethod
+    def __save_file(target_file_path: str, io_base_src: IOBase):
```

Comment:
@elegantmoose, relevant remark.

Actually, it would be good to use it but I cannot directly.
When defining the `post_payloads()` method (which indirectly calls `__save_file()`), I used the `@aiohttp_apispec.form_schema(PayloadCreateRequestSchema)` annotation in order to have a proper swagger form and documentation.

However, this annotation (hidden function) calls `request.multipart()` before I even enter the `post_payloads()` method, to check the form according to the given schema.
Then the file can be read from another special `request` dictionary object (see [aiohttp_apispec documentation](https://aiohttp-apispec.readthedocs.io/en/latest/usage.html#more-decorators)).
I cannot read twice the multipart form using standard *aiohttp* methods because the second time, I get no data.

Therefore, I cannot the `file_svc.save_multipart_file_upload()` or `file_svc.save_file()` directly because they assume nothing has called the `aiohttp` multipart methods yet.

However, I can try to do one of the following changes:

1. Adapt the current `file_svc.save_multipart_file_upload()` so that according to the given parameters, the file will be read the  *aiohttp* standard way or using a special *aiohttp_apispec* object produced by the `@aiohttp_apispec.form_schema` annotation,
2. Add another dedicated method to handle the *aiohttp_apispec* file upload case.

In both cases, I would try to use as maximum common code as possible.

Note that the `file_svc.save_multipart_file_upload()` method reads entirely the uploaded file and gives it as a `payload` parameter to `file_svc.save_file()`, which can lead to performance issues in the case of big files, but that's another story.

So tell me what you prefer (including another proposal).

Reply:
I like 2. 

But only if not a big lift. Can you take a quick look but cap your effort, dont want to hold this PR up.

Reply:
@elegantmoose, I had a look to the `file_svc.py` code.
It handles general file upload for other services or the (old v1?) API (I'm new to Cladera and I don't have the whole history).

Beside, this payload upload code looks the first API v2 file upload implementation (what is more, using *aiohttp_apispec* for swagger integration), and there are some global differences:

- `file_svc.py` (v1) handles file encoding (HTTP header) and encryption for storage as my v2 code doesn't,
- `file_svc.py` (v1) looks to handle some particular cases (file storage path, file extension), 
- The v2 code follows the *aiohttp_apispec* to handle uploaded file, so the way the file is read is different than in `file_svc.py` (however, the implementation I made should supports big file).

Finally, I'm running out of time for this work.

Therefore, I propose the following:

1. I only change the multipart form field name from *payload* to *file* to make my code as generic as possible,
2. If you agree, you merge this PR, and one nows that there is a base code for general API v2 file upload using *aiohttp_apispec*,
3. A future and distinct task / user story will handle a migration from v1 file upload to v2 in `file_svc.py`, with all features (file encoding, file encryption, *aiohttp_apispec* constraints for swagger).

Reply:
@elegantmoose, I pushed the field name change (see point 1. above).

I saw you merged from *master* in the *feature/manage-payloads* branch: I can rebase it (+ `git push --force`) to avoid a complex history if you want before validating this PR.

I'll create another PR for the GUI part (magma repository).

Reply:
Okay yea, too much of additional lift for this PR. Thanks for looking into though.

Yea, feel free to rebase it.

Im happy with this PR, I would just like @clenk to do quick look over.

Reply:
@elegantmoose, branch rebased and force pushed.